ibis/backends/dask/execution/reductions.py

```diff
@@ -73,37 +74,31 @@ def dask_execute_node_least_list(op, value, **kwargs):
     return compute_row_reduction(da.minimum, value)
 
 
-# TODO - aggregations - #2553
 @execute_node.register(ops.Reduction, ddgb.SeriesGroupBy, type(None))
 def execute_reduction_series_groupby(
     op, data, mask, aggcontext=None, **kwargs
 ):
     return aggcontext.agg(data, type(op).__name__.lower())
 
 
-def _filtered_reduction(mask, method, data):
-    return method(data[mask[data.index]])
+def _filtered_reduction(data, mask):
+    return make_selected_obj(data)[mask.obj].groupby(data.index)
 
 
-# TODO - aggregations - #2553
 @execute_node.register(ops.Reduction, ddgb.SeriesGroupBy, ddgb.SeriesGroupBy)
 def execute_reduction_series_gb_mask(
     op, data, mask, aggcontext=None, **kwargs
 ):
-    method = operator.methodcaller(type(op).__name__.lower())
-    return aggcontext.agg(
-        data, functools.partial(_filtered_reduction, mask.obj, method)
-    )
+    grouped_and_filtered_data = _filtered_reduction(data, mask)
```

Comment:
Why do we need to filter here for reduction?

Reply:
I'm not sure I understand the question?

We hit this operation when you're performing a reduction operation on a grouped series and your mask is also grouped. The pandas backend has a similar helper https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/execution/generic.py#L522 that works a bit differently.

Reply:
Aha I see. Unfortunately the pandas backend  code isn't well documented either :(

For my benefit, can you please explain what the mask is used for? Is it for the case where you have a where clause with the aggregation?

Reply:
I think https://github.com/ibis-project/ibis/pull/2596#discussion_r566222288 is the same question? Let me know if that still isn't clear.